contrib/mesos/pkg/scheduler/plugin.go

```diff
@@ -0,0 +1,875 @@
+		if err != nil {
+			log.Warningf("aborting Schedule, unable to understand pod object %+v", pod)
+			return "", noSuchPodErr
+		}
+		if deleted := k.podUpdates.Poll(podName, queue.DELETE_EVENT); deleted {
+			// avoid scheduling a pod that's been deleted between yieldPod() and Schedule()
+			log.Infof("aborting Schedule, pod has been deleted %+v", pod)
+			return "", noSuchPodErr
+		}
+		return k.doSchedule(k.api.tasks().Register(k.api.createPodTask(ctx, pod)))
+
+	//TODO(jdef) it's possible that the pod state has diverged from what
+	//we knew previously, we should probably update the task.Pod state here
+	//before proceeding with scheduling
+	case podtask.StatePending:
+		if pod.UID != task.Pod.UID {
+			// we're dealing with a brand new pod spec here, so the old one must have been
+			// deleted -- and so our task store is out of sync w/ respect to reality
+			//TODO(jdef) reconcile task
+			return "", fmt.Errorf("task %v spec is out of sync with pod %v spec, aborting schedule", task.ID, pod.Name)
+		} else if task.Has(podtask.Launched) {
+			// task has been marked as "launched" but the pod binding creation may have failed in k8s,
+			// but we're going to let someone else handle it, probably the mesos task error handler
+			return "", fmt.Errorf("task %s has already been launched, aborting schedule", task.ID)
+		} else {
+			return k.doSchedule(task, nil)
+		}
+
+	default:
+		return "", fmt.Errorf("task %s is not pending, nothing to schedule", task.ID)
+	}
+}
+
+// Call ScheduleFunc and subtract some resources, returning the name of the machine the task is scheduled on
+func (k *kubeScheduler) doSchedule(task *podtask.T, err error) (string, error) {
+	var offer offers.Perishable
+	if task.HasAcceptedOffer() {
+		// verify that the offer is still on the table
+		offerId := task.GetOfferId()
+		if offer, ok := k.api.offers().Get(offerId); ok && !offer.HasExpired() {
+			// skip tasks that have already have assigned offers
+			offer = task.Offer
+		} else {
+			task.Offer.Release()
+			task.Reset()
+			if err = k.api.tasks().Update(task); err != nil {
+				return "", err
+			}
+		}
+	}
+	if err == nil && offer == nil {
+		offer, err = k.api.algorithm()(k.api.offers(), k.api, task)
+	}
+	if err != nil {
+		return "", err
+	}
+	details := offer.Details()
+	if details == nil {
+		return "", fmt.Errorf("offer already invalid/expired for task %v", task.ID)
+	}
+	slaveId := details.GetSlaveId().GetValue()
+	if slave, ok := k.api.slaveFor(slaveId); !ok {
+		// not much sense in Release()ing the offer here since its owner died
+		offer.Release()
+		k.api.offers().Invalidate(details.Id.GetValue())
+		return "", fmt.Errorf("Slave disappeared (%v) while scheduling task %v", slaveId, task.ID)
+	} else {
+		if task.Offer != nil && task.Offer != offer {
+			return "", fmt.Errorf("task.offer assignment must be idempotent, task %+v: offer %+v", task, offer)
+		}
+		task.Offer = offer
+		task.FillFromDetails(details)
```

Comment:
OK, can you please add a comment here mentioning that? (Even better, referencing the open issue # specifically?)